internal/pkg/agent/install/install.go

```diff
@@ -44,18 +48,62 @@ func Install(cfgFile, topPath string, pt *progressbar.ProgressBar, streams *cli.
 		err = Uninstall(cfgFile, topPath, "", pt)
 		if err != nil {
 			pt.Describe("Failed to uninstall current Elastic Agent")
-			return errors.New(
+			return "", "", errors.New(
 				err,
 				fmt.Sprintf("failed to uninstall Agent at (%s)", filepath.Dir(topPath)),
 				errors.M("directory", filepath.Dir(topPath)))
 		}
 		pt.Describe("Successfully uninstalled current Elastic Agent")
 	}
 
+	uid := "0"
+	gid := "0"
+	username := ""
+	groupName := ""
+	if nonRoot {
+		username = elasticUsername
```

Comment:
nit: This block is big enough to be its own function.

Reply:
Okay will refactor it.